frappe/integrations/doctype/braintree_settings/braintree_settings.py

```diff
@@ -32,6 +32,9 @@ def on_update(self):
 		create_payment_gateway('Braintree-' + self.gateway_name, settings='Braintree Settings', controller=self.gateway_name)
 		call_hook_method('payment_gateway_enabled', gateway='Braintree-' + self.gateway_name)
 
+	def on_payment_request_submission(self, data):
```

Comment:
where do you want to use this? Why not just call it if `hasattr` is true? Adding an empty function in every payment settings is a chore and can be avoided?

Reply:
Hi @rmehta , you are entirely right. I have removed the methods and added `hasattr` in the payment request file. It's much cleaner.

Therefore I can close this PR, since no change is requested in Frappe.

Thanks a lot!